src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -834,7 +831,7 @@ private NamespaceOrTypeOrAliasSymbolWithAnnotations UnwrapAlias(NamespaceOrTypeO
             if (symbol.IsAlias)
             {
                 AliasSymbol discarded;
-                return NamespaceOrTypeOrAliasSymbolWithAnnotations.CreateUnannotated(NonNullTypesContext, (NamespaceOrTypeSymbol)UnwrapAlias(symbol.Symbol, out discarded, diagnostics, syntax, basesBeingResolved));
+                return NamespaceOrTypeOrAliasSymbolWithAnnotations.CreateUnannotated(symbol.IsNullableEnabled, (NamespaceOrTypeSymbol)UnwrapAlias(symbol.Symbol, out discarded, diagnostics, syntax, basesBeingResolved));
```

Comment:
>symbol.IsNullableEnabled [](start = 85, length = 24)

It would be good to update the speclet to cover aliases. 
My recollection is that the nullable context at the definition of the alias should not affect its top-level nullability, because it doesn't have one. The nullability comes from the place where the alias is used.
Is that correct?

Reply:
> It would be good to update the speclet to cover aliases. 

There is no change around aliases here, the context was taken from the binding position before and is taken from the binding position now. The ```symbol``` here is not an alias symbol, it is an NamespaceOrTypeOrAliasSymbolWithAnnotations structure. AliasSymbol stores only NamespaceOrTypeSymbol it never stored nullability or context. UsingDirectiveSyntax was never allowing NullableTypeSyntax as the target name.

---
In reply to: [239967172](https://github.com/dotnet/roslyn/pull/31607#discussion_r239967172) [](ancestors = 239967172)